system/addr.go

```diff
@@ -40,11 +40,16 @@ func (a *DefAddr) Exists() (bool, error) { return a.Reachable() }
 func (a *DefAddr) Reachable() (bool, error) {
 	network, address := splitAddress(a.address)
 
-	localAddr := &net.TCPAddr{
+	localTCPAddr, localUDPAddr := &net.TCPAddr{
+		IP: net.ParseIP(a.LocalAddress),
+	}, &net.UDPAddr{
 		IP: net.ParseIP(a.LocalAddress),
 	}
 
-	d := net.Dialer{LocalAddr: localAddr, Timeout: time.Duration(a.Timeout) * time.Millisecond}
+	d := net.Dialer{LocalAddr: localTCPAddr, Timeout: time.Duration(a.Timeout) * time.Millisecond}
+	if network == "udp" {
+		d = net.Dialer{LocalAddr: localUDPAddr, Timeout: time.Duration(a.Timeout) * time.Millisecond}
+	}
```

Comment:
Thinking the if condition should just be at the start. Instead of creating two variables.

```go
var localAddr net.Addr
if network == "udp" {
    localAddr = &net.UDPAddr{IP: net.ParseIP(a.LocalAddress)}
} else {
    localAddr = &net.TCPAddr{IP: net.ParseIP(a.LocalAddress)}
}
```

Reply:
Thanks for your advice. I refactor this part and do `make test-all` on my laptop